google/resource_container_cluster.go

```diff
@@ -1597,6 +1597,12 @@ func flattenClusterAddonsConfig(c *containerBeta.AddonsConfig) []map[string]inte
 				"disabled": c.KubernetesDashboard.Disabled,
 			},
 		}
+	} else {
```

Comment:
Unfortunately this is backwards - flatteners go from api objects to terraform objects, but expanders go from terraform objects to api objects.  You'll need to make a similar change in the expander.  Any reason you didn't choose to set the default in the schema object instead?